### PR TITLE
Update template further

### PR DESCRIPTION
**Pull request by chrisarevir**
Updates to the default template!

### CODE REVIEW

File: _template-particle/webpack.config.js [hunk 1]
```diff
@@ -1,54 +1,68 @@
...
-  ]
-});
+const path = require('path');
 
+const FriendlyErrorsWebpackPlugin = require('friendly-errors-webpack-plugin');
+const WebpackNotifierPlugin = require('webpack-notifier');
 
-const path = require('path');
+module.exports = () => {
+  const prod = process.env.NODE_ENV === 'production';
```
Comment (ben-burlingham) [inline _template-particle/webpack.config.js]: nb Parens?

File: _template-particle/package.json [hunk 2]
```diff
@@ -28,9 +28,11 @@
     "babel-plugin-transform-runtime": "^6.23.0",
     "babel-preset-env": "^1.6.1",
     "babel-preset-react": "^6.24.1",
+    "babel-preset-stage-2": "^6.24.1",
```
Comment (dancingshell) [inline _template-particle/package.json]: I thought we were getting rid of this and moving towards using the `.babelrc` with env?

```js
{
  "plugins": ["transform-runtime", "transform-export-extensions", "transform-class-properties"],
  "presets": ["env", "react"]
}
```

File: _template-particle/webpack.config.js (hunk 1 above)
Comment (chrisarevir) [reply _template-particle/webpack.config.js]: Not needed when it's a single statement and no ambiguous logic follows

File: _template-particle/webpack.config.js [hunk 3]
```diff
@@ -1,54 +1,68 @@
...
+
+    entry: './src/index.js',
+
+    module: {
+      rules: [
+        {
+          test: /\.jsx?$/,
+          loader: 'babel-loader',
+          include: [path.resolve(__dirname, 'src'), path.resolve(__dirname, 'sandbox')],
+          query: { plugins: ['transform-runtime'], presets: ['env', 'react', 'stage-2'] }
```
Comment (dancingshell) [inline _template-particle/webpack.config.js]: remove query?

File: _template-particle/webpack.config.js [hunk 4]
```diff
@@ -1,54 +1,68 @@
...
+        },
+        {
+          test: /\.s?[ca]ss$/,
+          exclude: /node_modules/,
+          use: [
+            { loader: 'style-loader' },
+            { loader: 'css-loader', options: { modules: true, importLoaders: 1, localIdentName: '[name]_[local]' } },
```
Comment (dancingshell) [inline _template-particle/webpack.config.js]: update to use new util https://github.com/procore/particles/blob/master/shared/cssUniqueNameBuilder.js

File: _template-particle/package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "dev": "NODE_ENV='development' webpack --progress --colors --watch",
     "prod": "NODE_ENV='production' webpack --progress --colors -p",
-    "sandbox": "NODE_ENV='development' webpack-dev-server --progress --colors --hot --inline ./sandbox/sandbox.jsx --config webpack.dev.config.js  --content-base ./../sandbox"
+    "sandbox": "NODE_ENV='development' webpack-dev-server --progress --colors --hot --inline ./sandbox/sandbox.jsx --content-base ./../sandbox"
```
Comment (dancingshell) [inline _template-particle/package.json]: yay one webpack file!

File: _template-particle/src/TemplateParticle.jsx [hunk 6]
```diff
@@ -3,18 +3,32 @@ import PropTypes from 'prop-types';
...
+    this.styles = composeStyles(baseStyles, props.stylesheets);
+
+    this.state = {
+      message: 'Your particle TemplateParticle says Hello World!'
+    };
+  }
+
+  render() {
+    return (
+      <div className={this.stylesheets.container}>
```
Comment (dancingshell) [inline _template-particle/src/TemplateParticle.jsx]: should be `this.styles.container`

File: _template-particle/package.json (hunk 2 above)
Comment (chrisarevir) [reply _template-particle/package.json]: Easier to just commit to going stage-2 as we should want things to work completely out of the box, and most projects benefit from using stage-2

File: _template-particle/src/TemplateParticle.jsx (hunk 6 above)
Comment (chrisarevir) [reply _template-particle/src/TemplateParticle.jsx]: Most likely, changing!

File: _template-particle/webpack.config.js (hunk 3 above)
Comment (chrisarevir) [reply _template-particle/webpack.config.js]: will do once .babelrc is in use!

File: _template-particle/webpack.config.js (hunk 4 above)
Comment (chrisarevir) [reply _template-particle/webpack.config.js]: On it!

File: _template-particle/package.json (hunk 2 above)
Comment (chrisarevir) [reply _template-particle/package.json]: Oh, and I couldn't get it to work on my side without explicitly installing stage-2 or other ones, so I think there's some magic going on with your setup that we should make explicit for all others so it works elsewhere as well.

Comment (dancingshell) [reply _template-particle/package.json]: ok that works

Comment (dancingshell) [reply _template-particle/package.json]: thats weird, i thought we were both able to get it to work the other day with the babelrc posted above. Ya let me know if you want to pair and get to the bottom of this

**Pull request merged**